src/classes/HiFiMixerSession.ts

```diff
@@ -358,39 +358,6 @@ export class HiFiMixerSession {
         let unGZippedData = pako.ungzip(data, { to: 'string' });
         let jsonData = JSON.parse(unGZippedData);
 
-        if (jsonData.deleted_visit_ids) {
-            let allDeletedUserData: Array<ReceivedHiFiAudioAPIData> = [];
-
-            let deletedVisitIDs = jsonData.deleted_visit_ids;
-            for (const deletedVisitID of deletedVisitIDs) {
-                let hashedVisitID = deletedVisitID;
-
-                let deletedUserData = new ReceivedHiFiAudioAPIData({
-                    hashedVisitID: hashedVisitID
-                });
-
-                let mixerPeerKeys = Object.keys(this._mixerPeerKeyToStateCacheDict);
-                for (const mixerPeerKey of mixerPeerKeys) {
-                    if (this._mixerPeerKeyToStateCacheDict[mixerPeerKey].hashedVisitID === hashedVisitID) {
-                        if (this._mixerPeerKeyToStateCacheDict[mixerPeerKey].providedUserID) {
-                            deletedUserData.providedUserID = this._mixerPeerKeyToStateCacheDict[mixerPeerKey].providedUserID;
-                        }
-                        // TODO: remove the entry from the peer state cache -- is this OK?
-                        //delete this._mixerPeerKeyToStateCacheDict[mixerPeerKey];
-                        break;
-                    }
-                }
-
-                allDeletedUserData.push(deletedUserData);
-            }
-
-            // TODO: remove the entry from the peer state cache
-            this.concurrency -= allDeletedUserData.length;
-            if (this.onUsersDisconnected && allDeletedUserData.length > 0) {
-                this.onUsersDisconnected(allDeletedUserData);
-            }
-        }
-
```

Comment:
I've moved the handling of **deleted_visit_ids** down below the handling of **peers**.  It is theoretically possible for a change to exist in peers for someone who was also deleted in the same frame, and when a change is not found in the map the peer is added to it.  By handling the deletions last we can avoid re-adding the changed user after deletion.